src/gen_rules.ml

```diff
@@ -689,18 +689,31 @@ Add it to your jbuild file to remove this warning.
       guess_modules ~dir:src_dir
         ~files:(Lazy.force files))
     in
-    List.filter_map stanzas ~f:(fun stanza ->
-      let dir = ctx_dir in
-      match (stanza : Stanza.t) with
-      | Library lib  ->
-        Some (library_rules lib ~dir
-                ~all_modules:(Lazy.force all_modules) ~files:(Lazy.force files)
-                ~scope)
-      | Executables  exes ->
-        Some (executables_rules exes ~dir ~all_modules:(Lazy.force all_modules)
-                ~scope)
-      | _ -> None)
-    |> Merlin.add_rules sctx ~dir:ctx_dir
+    let merlin =
+      List.filter_map stanzas ~f:(fun stanza ->
+        let dir = ctx_dir in
+        match (stanza : Stanza.t) with
+        | Library lib ->
+          Some (library_rules lib ~dir
+                  ~all_modules:(Lazy.force all_modules) ~files:(Lazy.force files)
+                  ~scope)
+        | Executables exes ->
+          Some (executables_rules exes ~dir ~all_modules:(Lazy.force all_modules)
+                  ~scope)
+        | _ -> None)
+      |> Merlin.merge_all in
+    Option.iter merlin ~f:(Merlin.add_rules sctx ~dir:ctx_dir);
+    Option.iter (Utop.exe_stanzas stanzas) ~f:(fun (exe, all_modules) ->
+      let dir = Utop.src_dir ctx_dir in
+      let merlin' = executables_rules exe ~dir ~all_modules ~scope in
+      Merlin.add_rules sctx ~dir merlin';
+      (* so that our uutop.ml module depends on the libraries that in our
+         current jbuild dir. We could also just pass the current jbuild dir
+         explicitly *)
+      let lib_requires =
```

Comment:
Do we really have to do this merge? The generated exe stanza already depends on all the libraries in the current dir, so using `merlin'.requires` should already contain everything we need.